src/drivers/shiftx_drv.c

```diff
@@ -67,8 +67,9 @@ void shiftx_handle_can_rx_msg(const CAN_msg *msg)
                 shiftx_update_config();
         }
 
-        if (msg->addressValue == shiftx_config.base_address + NOTIFICATION_BUTTON_STATE_OFFSET) {
-                pr_info_int_msg(_LOG_PFX "Received button event for base address: ", msg->addressValue);
+        if (msg->addressValue == shiftx_config.base_address + NOTIFICATION_BUTTON_STATE_OFFSET
+                        && shiftx_config.button_events_enabled) {
```

Comment:
Shouldn't this just omit generating the API event down below (lines 85-93)?  It would still be good to use the 'shiftx_rx_button_press' method to be able to query if a button was pressed on the ShiftX.

Reply:
Agreed. will change it.

Reply:
Ok, fixed.